File: LeetCode-Problems/Easy/Split-String-in-Balanced-String.py
# Python
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        balStr = 0
        for x in range(0, len(s)):
            if s[x:x+1] == "L":
                balStr = balStr + 1
            else:
                balStr = balStr - 1
            if balStr == 0:
                count = count + 1
        return count

File: LeetCode-Problems/Easy/Split-String-in-Balanced-String.java
//Java
class Solution {
    public int balancedStringSplit(String s) {
        int count = 0;
        int balStr = 0;
        for (int i = 0; i < s.length(); i++){
            if (s.charAt(i) == 'L'){
                balStr++;
            }
            else{
                balStr--;
            }
            if (balStr == 0){
                count++;
            }
        }
        return count;
    }
}
